talk/objectorientation/advancedoo.tex

```diff
@@ -1,5 +1,44 @@
 \subsection[advOO]{Advanced OO}
 
+\begin{frame}[fragile]
+  \frametitlecpp[98]{\texttt{this} keyword}
+  \begin{block}{How to know an object's address?}
+    \begin{itemize}
+    \item Sometimes we need the address of the current object
+    \item Or we need to pass our address / a reference to a different entity
+      (for example to implement operators, see later)
+    \item All class methods can use the keyword \cppinline{this}
+      \begin{itemize}
+        \item It returns the address of the current object
+        \item Its type is \cppinline{T*} in the methods of a class {\ttfamily T}
```

Comment:
For `const` member functions it would be `const T*`. But you can just mention that.

Reply:
You wrote that line initially, so I'm happy that you are still almost happy with it. 🙂

Reply:
:D